rhea/build/boards/altera/_de1_SoC.py

```diff
@@ -0,0 +1,229 @@
+#
+# Copyright (c) 2015 Christopher L. Felton
+#
+# 
+# Added Support for the Terasic De1-SoC Board
+# By Mike Kennedy (Va3TeC)
+#
+
+from __future__ import absolute_import
+
+from string import Template
+
+from rhea.build import FPGA
+from rhea.build.toolflow import Quartus
+
+
+class DE1SoC(FPGA):
+    vendor = 'altera'
+    family = 'Cyclone V'
+    device = '5CSEMA5F31C6'
+    speed = '6'
+    _name = 'de1_SoC'
+
+
+# As Per page 23 of the user Guide for DE1-SoC
+    default_clocks = {
+        'clock': dict(frequency=50e6, pins=('AF14',)) 
+        #'clock2': dict(frequency=50e6, pins=('AA16',))
+        #'clock3': dict(frequency=50e6, pins=('Y26',))
+        #'clock4': dict(frequency=50e6, pins=('K14',))
+    }
+
+    default_resets = {
+        'reset': dict(active=0, async=True, pins=('',)) #Fixc
+    }
+    
+    default_ports = {
+        'led': dict(pins=('V16', 'W16', 'V17', 'V18', 'W17', 'W19', 'Y19', 'W20', 'W21', 'Y21')),
+        'sw_key': dict(pins=('AA14', 'AA15', 'W15', 'Y16')),
+        'sw_slide': dict(pins=('AB12', 'AC12', 'AF9', 'AF10','AD11','AD12','AE11','AC9','AD10','AE12')),
+	
+	'7-seg_0': dict(pins=(	'AE26', # HEX0[0]
+				'AE27', # HEX0[1]      
+				'AE28', # HEX0[2]      
+				'AG27', # HEX0[3]      
+				'AF28', # HEX0[4]      
+				'AG28', # HEX0[5]      
+				'AH28', # HEX0[6]      
+				)),	
+	
+	'7-seg_1': dict(pins=(	'AJ29', # HEX1[0]
+				'AH29', # HEX1[1]      
+				'AH30', # HEX1[2]      
+				'AG30', # HEX1[3]      
+				'AF29', # HEX1[4]      
+				'AF30', # HEX1[5]      
+				'AD27', # HEX1[6]      
+				)),	
+	
+	'7-seg_2': dict(pins=(	'AB23', # HEX2[0]
+				'AE29', # HEX2[1]      
+				'AD29', # HEX2[2]      
+				'AC28', # HEX2[3]      
+				'AD30', # HEX2[4]      
+				'AC29', # HEX2[5]      
+				'AC30', # HEX2[6]      
+				)),	
+	
+	'7-seg_3': dict(pins=(	'AD26', # HEX3[0]
+				'AC27', # HEX3[1]      
+				'AD25', # HEX3[2]      
+				'AC25', # HEX3[3]      
+				'AB28', # HEX3[4]      
+				'AB25', # HEX3[5]      
+				'AB22', # HEX3[6]      
+				)),	
+	
+	'7-seg_4': dict(pins=(	'AA24', # HEX4[0]
+				'Y23', # HEX4[1]      
+				'Y24', # HEX4[2]      
+				'W22', # HEX4[3]      
+				'W24', # HEX4[4]      
+				'V23', # HEX4[5]      
+				'W25', # HEX4[6]      
+				)),
+	
+	'7-seg_5': dict(pins=(	'V25', # HEX5[0]
+				'AA28', # HEX5[1]      
+				'Y27', # HEX5[2]      
+				'AB27', # HEX5[3]      
+				'AB26', # HEX5[4]      
+				'AA26', # HEX5[5]      
+				'AA25', # HEX5[6]      
+				)),	
+
+
+        # bi-directional GPIO, 
+        # @todo: finish the GPIO pins
+        'gpio': dict(pins=(     'AC18', # GPIO_0[0]
```

Comment:
Don't think we need a comment on each line, if we have 4-6 pin defs per line, it will be more compact and readable, simply use a single comment on each line  `# 0-5` to indicate with logical GPIO indexes were defined.